### ISSUE TITLE
No notification is shown when executing `su -c 'PATH=/system/bin:/system/xbin id'`

### ISSUE THREAD

**Issue by Henry-ZHR** [topjohnwu/Magisk#5302]
When I execute `su -c 'PATH=/system/bin:/system/xbin id'`, there is no notification saying that `xxx was granted Superuser rights`; `/system/bin:/system/xbin` can be replaced with something like `/sbin:/system/xbin` or even `/abc:/def`

But when I execute any of the following, the notification works:
```shell
su -c 'PATH=/system/bin id' # The output is 'uid=0(root) gid=0(root) groups=0(root) context=u:r:magisk:s0', so I believe it does get root access
su -c 'PATH=/abc id' # /abc doesn't exist; 'inaccessible or not found' is reported, but this is expected
```

Device: 
Android version: 12
Magisk version name: e8f31c78
Magisk version code: 24001

Log of excuting `adb logcat -c && sleep 1 && adb shell "su -c 'PATH=/xxx:/system/bin id'" && sleep 1 && adb logcat -d > log0.txt`: https://gist.github.com/Henry-ZHR/ab9d244e1d34e09829c2d1a29adc625a; nothing seems useful

**Comment by yujincheng08** [comment]
The bug is from the following line. The buffer is not large enough and does not even use `snprintf`:

https://github.com/topjohnwu/Magisk/blob/ea75a09f95bb6d3117f4050f0c99266c007edaf6/native/jni/su/connect.cpp#L84

**Comment by topjohnwu** [comment]
Fixed in a4f97fa1513e7824b4b4b5a95dbf20969ec3a845

**Comment by yujincheng08** [comment]
Fixed by a4f97fa
